test/utils/math/Math.t.sol

```diff
@@ -206,23 +206,43 @@ contract MathTest is Test {
         } catch {}
     }
 
+    // MOD EXP
+    function testModExp(uint256 b, uint256 e, uint256 m) public {
+        try this.modexp(b, e, m) returns (uint256 result) {
+            assertTrue(result < m);
+            assertEq(result, _nativeModExp(b, e, m));
+        } catch {
+            assertEq(m, 0);
+        }
+    }
+
+    function _nativeModExp(uint256 b, uint256 e, uint256 m) private pure returns (uint256) {
+        uint256 r = 1 % m;
+        while (e > 0) {
```

Comment:
```suggestion
        uint256 r = 1 % m;
        if(r == 0) return 0;
        while (e > 0) {
```

Reply:
I'm not sure I understand that change.

The only way for `r` to be 0 is for m to be `1`.
In that case, then everything is 0, all operations done mod m return 0.
This means the while loop will try to do stuff, but in the end, we will have `r=0`.

I'm not sure what we are trying to optimise. Its not production code, its a reference for fuzzing.

Reply:
Not trying to optimize, just thought it was better to avoid the full computation and return right ahead.

I'm fine with the changes you made fixing `r` to 1